### DISCUSSION TITLE
Can't Remove File GH web Using: git rm (file name) or git rm --cached (file name))

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22829] [New to GitHub]
<p>I have gotten this message repeatedly and do not understand.<br>
“not removing ‘blog’ recursively without -r”</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I assume <code>blog</code> is a directory? Git is telling you that it won’t remove a directory (and thus <em>recursively</em> all its content) unless you explicitly tell it to remove recursively. Either give <code>git rm</code> specific <strong>files</strong>, or if you really want to remove the directory and everything in it use the <code>-r</code> flag.</p>

**Comment by ghost** [comment]
<p>Did I enter this incorrectly?<br>
<img src="https://user-images.githubusercontent.com/70776696/181073276-18252fb6-5aa5-4b60-a724-8ba2f4e1b6f4.png" alt="Capture8" data-base62-sha1="mc08UBXyDHeIXn8yQluCQPJRAwF" width="654" height="170"></p>

**Comment by airtower-luna** [comment]
<p><code>-r</code> is a flag to <code>git rm</code>, so you’d have to use something like <code>git rm -r blog</code>. The second version is almost correct, but with a <code>=</code> instead of <code>-</code>.</p>

**Comment by ghost** [comment]
<p>Awesome, thank you, I was successful</p>

**Comment by Princelion** [comment]
You also can use "git rm -r --cached <file name> " which worked smoothly as well.
